### PR TITLE
EOS-16890:Support for get_all_keys in S3confstore

### PR DESCRIPTION
**Pull request by AtitaShirwaikar**
**### Issue**
Cortx-Utils had support for get_keys() which lists all the key-value pairs in a file.
Hooked up the support in s3confstore for get_keys API

### **Testing:**
[root@ssc-vm-c-1632 cortx-s3server]# s3confstore yaml:///opt/seagate/cortx/s3/s3backgrounddelete/config.yaml getallkeys

['version_config>version', 'cortx_s3>endpoint', 'cortx_s3>service', 'cortx_s3>default_region', 'cortx_s3>daemon_mode', 'cortx_s3>s3_instance_count', 'cortx_s3>messaging_platform', 'message_bus>topic', 'message_bus>consumer_group', 'message_bus>consumer_id_prefix', 'message_bus>producer_id', 'message_bus>producer_delivery_mechanism', 'message_bus>consumer_sleep', 'logconfig>scheduler_logger_name', 'logconfig>processor_logger_name', 'logconfig>logger_directory', 'logconfig>scheduler_log_file', 'logconfig>processor_log_file', 'logconfig>file_log_level', 'logconfig>console_log_level', 'logconfig>log_format', 'logconfig>max_bytes', 'logconfig>backup_count', 'rabbitmq>username', 'rabbitmq>password', 'rabbitmq>host', 'rabbitmq>queue', 'rabbitmq>exchange', 'rabbitmq>exchange_type', 'rabbitmq>mode', 'rabbitmq>durable', 'rabbitmq>schedule_interval_secs', 'indexid>probable_delete_index_id', 'indexid>global_instance_index_id', 'indexid>global_bucket_index_id', 'indexid>bucket_metadata_index_id', 'indexid>max_keys', 'leakconfig>leak_processing_delay_in_mins', 'leakconfig>version_processing_delay_in_mins', 'leakconfig>cleanup_enabled']

### **UT Result**
Running s3confstore UT's...
======================================== test session starts ========================================
platform linux -- Python 3.6.8, pytest-2.9.2, py-1.4.32, pluggy-0.3.1 -- /usr/bin/python36
cachedir: ../.cache
rootdir: /workspace/cortx-s3server/s3cortxutils/s3confstore, inifile:
collected 19 items

../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_get_nodecount_none PASSED
../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_get_nodecount_success PASSED
../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_get_nodenames_list_empty PASSED
../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_get_nodenames_list_exception PASSED
../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_get_nodenames_list_success PASSED
../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_get_privateip_emptystring PASSED
../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_get_privateip_success PASSED
../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_get_publicip_emptystring PASSED
../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_get_publicip_success PASSED
../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_get_s3instance_count_success PASSED
../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_json_conf PASSED
../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_mock_get PASSED
**../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_mock_getallkeys PASSED**
../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_mock_load PASSED
../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_mock_save PASSED
../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_mock_set PASSED
../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_validate_configfile_doesnotexist PASSED
../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_validate_configfile_unsupportedformat PASSED
../ut/s3confstore_unit_tests.py::S3ConfStoreAPIsUT::test_yaml_conf PASSED

===================================== 19 passed in 0.11 seconds =====================================

### CODE REVIEW

**Review by nileshgovande — APPROVED** [review]

**Review by vimalraghwani20 — APPROVED** [review]

File: s3cortxutils/s3confstore/s3confstore/cortx_s3_confstore.py [hunk 1]
```diff
@@ -54,6 +54,10 @@ def set_config(self, key: str, value: str, save: bool = False):
       # Update the index backend.
       Conf.save(self.default_index)
 
+  def get_all_keys(self):
```
Comment (amitwac2608) [inline s3cortxutils/s3confstore/s3confstore/cortx_s3_confstore.py:57]: Please write UT for this function

**Review by amitwac2608 — CHANGES_REQUESTED** [review]

**Comment by gowthamchinna** [comment]
run mini-provisioning

File: s3cortxutils/s3confstore/s3confstore/cortx_s3_confstore.py (hunk 1 above)
Comment (AtitaShirwaikar) [reply s3cortxutils/s3confstore/s3confstore/cortx_s3_confstore.py:57]: done

**Comment by nileshgovande** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/620/

**Comment by nileshgovande** [comment]
MIni provisioning passed http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Mini-Provisioning/job/cortx-s3server/216/
Pre - merge #620 also passed. Merging

**Pull request merged**